### PR TITLE
[enhancement] Re-add spaces to setting names 

### LINKED ISSUE #1813: Re-add spaces to setting names
#972 Since we have greedy string already. What is the point of having weird user unfriendly name with no spaces.

:wonk:

### PR DESCRIPTION
**Pull request by scorbett123**
Re-add spaces to setting names, if you attempt to toggle with chat you can just do it without spaces:
e.g.
toggling all entities in chams is done through #set chams allEntities toggle.

closes #1813.
Also improved on the grammar changing all entity to all entities.

### CODE REVIEW

**Comment by l1ving** [comment]
> if you attempt to toggle with chat you can just do it without spaces

This is unintuitive imo, I'm not sure if this is a good idea. I mean I prefer it to `;set module "setting name" value` but it could be confusing.

**Comment by scorbett123** [comment]
> This is unintuitive imo, I'm not sure if this is a good idea. I mean I prefer it to `;set module "setting name" value` but it could be confusing.

I understand, but I feel like now there are string settings in the gui, making the gui look nice should take priority over making settings through chat more intuitive because most people will rarely use them.

**Comment by l1ving** [comment]
cries in cli user

**Comment by scorbett123** [comment]
Do you want me to make it so that you can do either?

**Comment by l1ving** [comment]
Nah this is good, I'll review later

**Comment by Xiaro** [comment]
> > if you attempt to toggle with chat you can just do it without spaces
> 
> This is unintuitive imo, I'm not sure if this is a good idea. I mean I prefer it to `;set module "setting name" value` but it could be confusing.

find settings with both space included and space excluded lol

**Comment by nopjmp** [comment]
It might be better to change the space to an underscore so it's still readable in the chat cli.

**Comment by l1ving** [comment]
when taking setting names you could just remove the _ and spaces before finding the setting

**Comment by Xiaro** [comment]
> It might be better to change the space to an underscore so it's still readable in the chat cli.

Upper case words and formatting will make it readable and the name without space won't be used for displaying. Only for looking up in set command

**Comment by scorbett123** [comment]
Is this better? Also, on a separate note, what is the difference between a greedy (string) and a normal string?

**Comment by Xiaro** [comment]
> Is this better? Also, on a separate note, what is the difference between a greedy (string) and a normal string? I know that you will complain about the formatting of the getSetting function, however faced with that I have no idea what to do.

Normal string take only one piece of argument, either surrounded by space ` ` or quote `"`. Greedy will all the input to the end so you can only use it at the end of an argument tree (right before the execute)

**Comment by scorbett123** [comment]
> > Is this better? Also, on a separate note, what is the difference between a greedy (string) and a normal string? I know that you will complain about the formatting of the getSetting function, however faced with that I have no idea what to do.
> 
> Normal string take only one piece of argument, either surrounded by space ` ` or quote `"`. Greedy will all the input to the end so you can only use it at the end of an argument tree (right before the execute)

Thanks

File: src/main/kotlin/org/kamiblue/client/command/commands/SetCommand.kt [hunk 1]
```diff
@@ -90,15 +105,21 @@ object SetCommand : ClientCommand(
                 val module = moduleArg.value
                 val settingList = module.fullSettingList
 
-                MessageSendHelper.sendChatMessage("List of settings for ${formatValue(module.name)}: " +
-                    formatValue(settingList.size)
-                )
-                MessageSendHelper.sendRawChatMessage(settingList.joinToString { it.name })
+                MessageSendHelper.sendChatMessage("List of settings for ${formatValue(module.name)}: ${formatValue(settingList.size)}")
+                MessageSendHelper.sendRawChatMessage(settingList.joinToString { "${it.name.removeSpaceAndLowerCase()}(${it.name})" })
```
Comment (l1ving) [inline src/main/kotlin/org/kamiblue/client/command/commands/SetCommand.kt:109]: Why are we printing settings with lowercase in the chat? It's less readable

File: src/main/kotlin/org/kamiblue/client/command/commands/SetCommand.kt [hunk 2]
```diff
@@ -90,15 +105,21 @@ object SetCommand : ClientCommand(
                 val module = moduleArg.value
                 val settingList = module.fullSettingList
 
-                MessageSendHelper.sendChatMessage("List of settings for ${formatValue(module.name)}: " +
-                    formatValue(settingList.size)
-                )
-                MessageSendHelper.sendRawChatMessage(settingList.joinToString { it.name })
+                MessageSendHelper.sendChatMessage("List of settings for ${formatValue(module.name)}: ${formatValue(settingList.size)}")
```
Comment (l1ving) [inline src/main/kotlin/org/kamiblue/client/command/commands/SetCommand.kt:108]: I don't think the setting list size should be including fullSettingList, which has the module enabled state and visible state, and other stuff

**Review by l1ving — CHANGES_REQUESTED** [review]

File: src/main/kotlin/org/kamiblue/client/command/commands/SetCommand.kt (hunk 2 above)
Comment (Xiaro) [reply src/main/kotlin/org/kamiblue/client/command/commands/SetCommand.kt:108]: I don't really know why I added that neither. 🗿

Comment (l1ving) [reply src/main/kotlin/org/kamiblue/client/command/commands/SetCommand.kt:108]: I mean setting list size is fine, but it should be normal setting list, not full

Comment (Xiaro) [reply src/main/kotlin/org/kamiblue/client/command/commands/SetCommand.kt:108]: > I don't really know why I added that neither. 🗿

Uhh replied to the wrong one. Was for "Why are we printing settings with lowercase in the chat? It's less readable"

Comment (Xiaro) [reply src/main/kotlin/org/kamiblue/client/command/commands/SetCommand.kt:108]: > I don't think the setting list size should be including fullSettingList, which has the module enabled state and visible state, and other stuff

No, set command should allow you change all setting including visible, invisible ones. And it is only 3 or 4 settings

Comment (l1ving) [reply src/main/kotlin/org/kamiblue/client/command/commands/SetCommand.kt:108]: Alright fine. Just the lowercase when printing is dumb

**Review by l1ving — APPROVED** [review]

**Comment by l1ving** [comment]
Thank you!

**Comment by scorbett123** [comment]
Thanks.

**Pull request merged**